### DISCUSSION TITLE
problem when cutting png 48bits deep

### DISCUSSION THREAD

**Discussion by Jupklass** [ImageMagick/ImageMagick#7657] [Help]
### ImageMagick version

8.0.0

### Operating system, version and so on

c# .net8 windows 11

### Description

I have a piece of code that works fine with all images to split the image in multiple subimages.
but it fails with png of 48bits deep (same png but with 24 bits do it fine)

if in geometry i multiply x axis  *2  it does it rigth
but there is no way to fix  Y axis  
any idea what is happening?
thanks

Code -->

/// with this 5 lines works the x axe of png 48 bits
 if (Path.GetExtension(ImagePath).ToLower() == ".png")
 {
     int ModifyCut= Convert.ToInt32(Original_Image.Depth) / 8;
     Cut.X *= ModifyCut;
 }

// "standar Code"

 MagickGeometry Geometria = new()
 {
     Width = Convert.ToUInt32(Cut.X),
     Height = Convert.ToUInt32(Cut.Y),
     X = 0,
     Y = 0,
     IgnoreAspectRatio = true
 };
            

 MagickImage Lateral = (MagickImage)Original_Image.Clone();
 Lateral.Crop(Geometria);

Original image [ you can download at https://mega.nz/file/cVkEkDiQ#wLACypGE6jqe-rTgFTycSGVGS6LbWFYYWJ0xds53tNI

![Original2](https://github.com/user-attachments/assets/bcf9b07e-6f37-4255-8852-5c7a50c8de50)

splited image x and y failure on 48 bits deep

![Captura de pantalla 2024-10-01 151310](https://github.com/user-attachments/assets/91547f37-7cb7-473b-a071-c865c5cf432e)

splited image x axis fixed (2*) but y still failure

![Captura de pantalla 2024-10-01 151515](https://github.com/user-attachments/assets/f1e0504a-e1b3-49a9-956d-1a377a667549)

**Comment by snibgo** [comment]
Naming your first image as "tracks.png":
```
magick tracks.png info:
magick: bKGD: invalid color `tracks.png' @ warning/png.c/MagickPNGWarningHandler/1526.
tracks.png PNG 247x1014 2480x3508+387+64 8-bit TrueColorAlpha sRGB 492366B 0.000u 0:00.000
```
So your image has a virtual canvas, which complicates the crop operation. I suggest you do the equivalent of "+repage" immediately after reading the input.

**Reply by Jupklass** [reply]
Thanks
I will try tomorrow and i will let you know

**Reply by Jupklass** [reply]
Thank you !!
it works!!

just add   Lateral.ResetPage();  

 after MagickImage Lateral = (MagickImage)Imagen_a_cortar.Clone();
